test/e2e/run

```diff
@@ -130,8 +131,27 @@ def set_jobs(litani, run_dir, mod):
 def run_build(litani, run_dir, mod):
     os.chdir(run_dir)
     args = mod.get_run_build_args()
-    run_litani(
-        litani, "run-build", *args.get("args", []), **args.get("kwargs", {}))
+    proc = run_litani(
+        litani, "run-build",
+        *args.get("args", []), check=False, **args.get("kwargs", {}))
+
+    try:
+        expected_rc = mod.get_run_build_return_code()
+        if any((
```

Comment:
This does not test that expected_rc == proc.returncode as your commit messages says it will.

Reply:
Does this address the customer problem motivating this change?  I thought the customer concern was proof failure did not cause the run script to fail in the customer context (while it should not cause the run script to fail in the standard continuous integration context).

Reply:
Done, thank you.

Yes, the run script wasn't failing because Litani wasn't, the run script was just checking Litani's return code rather than reading run.json. This fixes that bug.